src/portable/microchip/samd/dcd_samd.c

```diff
@@ -183,9 +183,12 @@ void dcd_connect(uint8_t rhport)
 void dcd_sof_enable(uint8_t rhport, bool en)
 {
   (void) rhport;
-  (void) en;
 
-  // TODO implement later
+  if (en) {
+    USB->DEVICE.INTENSET.bit.SOF = 1;
+  } else {
+    USB->DEVICE.INTENCLR.bit.SOF = 1; 
+  }
```

Comment:
SAMD support has been tested (both enable and disable) on a [Seeed Studio XIAO SAMD21](https://wiki.seeedstudio.com/Seeeduino-XIAO-TinyUSB/)